### ISSUE TITLE
Multiple customtags fail validation

### ISSUE THREAD

**Issue by michaelpollak** [dasistwas/moodle-mod_datalynx#96]
When I add a field or custom tag (eg ##viewsmenu##) multiple times I get duplicate IDs. We could prevent users from doing this but sometimes it makes sense to be redundant.

"Tags like ##viewsmenu## can be added more than once. So if we assign an id to those elements we should assign also a number, that indicates the first, second, thirds... appearance. But I think for elements, that appear more than once, maybe we can omit the id, and just add a class and maybe some data attribute. Not sure what is the benefit of adding ids, except selecting the elements efficiently with JS. But that seems not always to be the case: https://stackoverflow.com/questions/31026901/processing-speed-difference-between-css-class-and-id So my preferred option would be to get rid of the id."

**Comment by dasistwas** [comment]
Yes, if ids are not used, than get rid of them and replace with classes.

**Comment by michaelpollak** [comment]
I have worked on this some more and feel a solution is nearly impossible.
The ids can be changed but not entirely removed. It is possible to make an algorithmic solution that renames every iteration but this will need to also rename the input field. This does not make sense as inputs will be lost. 
In a way I feel like either multiple uses of the same tag should not be allowed or the html validation errors should be ignored. As we discussed multiple uses of the same tag in principle should be allowed ignoring validation errors in my mind is the only sensible option we have in this case.

**Comment by dasistwas** [comment]
There is a difference between viewmode and edit mode (Moodle forms,). For edit mode it does not make sense to have a tag two times, if it is editable. Only in view mode it would make sense.

**Comment by michaelpollak** [comment]
I've worked on this today and have no idea what else to try :) 
I'd call this "okayish" and close the issue if you agree.
